services/app-api/src/resolvers/healthPlanPackage/submitHealthPlanPackage.test.ts

```diff
@@ -135,6 +139,172 @@ describe.each(flagValueTestParameters)(
             ).toBeGreaterThan(0)
         }, 20000)
 
+        it('returns a state submission with the correct rate data on resubmit', async () => {
+            const cmsUser = testCMSUser()
+            const server = await constructTestPostgresServer({
+                ldService: mockLDService,
+            })
+
+            const cmsServer = await constructTestPostgresServer({
+                context: {
+                    user: cmsUser,
+                },
+                ldService: mockLDService,
+            })
+
+            const initialRateInfos = () => ({
+                id: uuidv4(),
+                rateType: 'NEW' as const,
+                rateDateStart: new Date(Date.UTC(2025, 5, 1)),
+                rateDateEnd: new Date(Date.UTC(2026, 4, 30)),
+                rateDateCertified: new Date(Date.UTC(2025, 3, 15)),
+                rateDocuments: [
+                    {
+                        name: 'rateDocument.pdf',
+                        s3URL: 'fakeS3URL',
+                        sha256: 'fakesha',
+                        documentCategories: ['RATES' as const],
+                    },
+                ],
+                supportingDocuments: [],
+                rateProgramIDs: [defaultFloridaRateProgram().id],
+                actuaryContacts: [
+                    {
+                        name: 'test name',
+                        titleRole: 'test title',
+                        email: 'email@example.com',
+                        actuarialFirm: 'MERCER' as const,
+                        actuarialFirmOther: '',
+                    },
+                ],
+                actuaryCommunicationPreference: 'OACT_TO_ACTUARY' as const,
+                packagesWithSharedRateCerts: [],
+            })
+
+            // First, create new submissions
+            const submittedEditedRates =
+                await createAndSubmitTestHealthPlanPackage(server, {
+                    rateInfos: [initialRateInfos()],
+                })
+            const submittedNewRates =
+                await createAndSubmitTestHealthPlanPackage(server, {
+                    rateInfos: [initialRateInfos()],
```

Comment:
This is creating rates in two submissions with the same ID, is that intentional? That's not something we expect to have ever happened in our app

Reply:
I don't think that's what is happening this is a function call and its using uuid util - I would think they have different ids

Reply:
> I don't think that's what is happening this is a function call and its using uuid util - I would think they have different ids

Yes!

Reply:
right!